ulislaw tikers sleep

mprse opened the pull request:
## Description

Provide the following changes:
- Disable us ticker in deep-sleep mode if running (make it consistent with sleep requirements). Verified that hal sleep test from feature-hal-spec-sleep branch passes.
- Update tests-mbed_drivers-timerevent(see commit comment).
- Modify tolerance in tests-mbed_drivers-lp_ticker (see commit comment).
- Enable all NRF51 boards which are enabled on current master (previous version was invalid and PR has been merged before I pushed the fix).

## Status
**READY**


## Migrations
NO

mprse committed:
tests-mbed_drivers-timerevent: Disable deep-sleep mode in TestTimerEvent class.

Since now us ticker is disabled in deep-sleep mode and tests-mbed_drivers-timerevent verifies TimerEvent class which uses us ticker then deep-sleep mode needs to be disabled during the test execution.

mprse committed:
tests-mbed_drivers-lp_ticker: increase tolerance for NRF51_DK

This test fails sometimes while testing 1 ms timeout on NRF51_DK board.
This board is very slow and lp ticker is inaccurate.
I noticed that measured results are very close to the tolerance value. For 1ms delay we measure in failure case 1560 us and tolerance is set to 550.
Since measurement method is also significantly flawed (extra time for setting up ticker and execution of a ticker irq handler) I recommend to increase tolerance by least +100 us for now.

mprse committed:
Disable us ticker in deep-sleep mode.

mprse committed:
Enable support for NRF51 boards.

Make it consistent with master.

bulislaw commented on TESTS/mbed_drivers/timerevent/main.cpp:
```diff
@@ -39,13 +39,16 @@ class TestTimerEvent: public TimerEvent {
     TestTimerEvent() :
             TimerEvent(), sem(0, 1) {
+
+        sleep_manager_lock_deep_sleep_internal();
```
You shouldn't use internal versions, but the public macro.

mprse replied:
Fixed.

mprse committed:
tests-mbed_hal-common_tickers: disable os tick while running test cases

It has been noticed on NRF51_DK board that occurrence of system tick have impact on test execution.
NRF51_DK is a slow board with fast us ticker and handling of system tick interrupt takes about 250 us ticker ticks which have huge influence on interrupt tests where interrupt is set to <current count> + 100, 200, 500 ticks.

mprse commented:
Fixed after review.
Added fix to test `tests-mbed_hal-common_tickers` (disable system tick while running test cases).
Successfully executed all tests on NRF51_DK/GCC_ARM.
